src/codegen/yul/builtin.rs

```diff
@@ -122,23 +125,43 @@ pub(crate) fn process_builtin(
         | YulBuiltInFunction::CallCode
         | YulBuiltInFunction::DelegateCall
         | YulBuiltInFunction::StaticCall
-        // Return and revert also load from memory, so we first need to solve mload and mstore builtins
-        | YulBuiltInFunction::Return
-        | YulBuiltInFunction::Stop // Stop is the same as return(0, 0)
-        | YulBuiltInFunction::Revert
         // Log functions
         | YulBuiltInFunction::Log0
         | YulBuiltInFunction::Log1
         | YulBuiltInFunction::Log2
         | YulBuiltInFunction::Log3
         | YulBuiltInFunction::Log4
         // origin is the same as tx.origin and is not implemented
-        | YulBuiltInFunction::Origin
-        => {
+        | YulBuiltInFunction::Origin => {
             let function_ty = builtin_ty.get_prototype_info();
             unreachable!("{} yul builtin not implemented", function_ty.name);
         }
 
+        // Return and revert also load from memory, so we first need to solve mload and mstore builtins
+        YulBuiltInFunction::Return => {
+            if ns.target != Target::EVM {
+                let function_ty = builtin_ty.get_prototype_info();
+                unreachable!("{} yul builtin not implemented", function_ty.name);
+            }
+
+            let addr = expression(&args[0], contract_no, ns, vartab, cfg, opt).cast(&Type::Ref(Type::Uint(256).into()), ns);
+            let length = expression(&args[0], contract_no, ns, vartab, cfg, opt).cast(&Type::Uint(256), ns);
+
+            cfg.add(vartab, Instr::Return { value: vec![addr, length] });
```

Comment:
I am not sure this is the correct return values. According to [Solidity documentation](https://docs.soliditylang.org/en/latest/yul.html#evm-dialect), the returned value is not the address and the length but `mem[p…(p+s))`. You are not loading the data for returning. 

`return(p, s) => end execution, return data mem[p…(p+s)]`

Reply:
In addition, these changes are not tested in any runtime test.

Reply:
We don't have EVM code gen so these cannot be tested. These were simply added to make the `fn ethereum_solidity_tests()` work.

Reply:
Not only the implementation is not following the documentation, but also we cannot test if this indeed works. I don't like the idea of adding code that we cannot ensure it is working. Why don't we raise a warning in sema when these buildings are used? This way, we don't go through this code path.

Also, why do we still have codegen for EVM? Maybe it is time to disable this.

Reply:
Adding EVM support is hugely beneficial to solang. 
* We can run the eth solidity semantic tests through sema and codegen which is yielding an enormous amount of issues in our code. This PR would not exist if it weren't for those tests.
* You can see that [solang-parser](https://crates.io/crates/solang-parser) is quite poplar because it can parse eth solidity. If sema could also work with eth solidity, then we could publish `solang-sema` and that crate would get users, contributors, testers, and bug reports
* Our language server would be much more popular if we have better evm support
* It would be great if Solang had EVM support including codegen and emit, and could create contracts which can execute on EVM chains. This would require a lot of work and is not something we would directly spend time on, but we can do mentorships and such things.

Yes, this particular bit of code path is not tested, but we need to run evm contracts through codegen to get all the warnings/errors. The problems with evm codegen will hopefully be fixed later. In the mean time, this is sufficient for our purposes (getting all diagnostics and ast for evm solidity contracts).

Reply:
I suggest you to at least implement the builtin correctly then.
Adding a large comment explaining that the implementation is not correct and should be fixed would also help.

Reply:
> I suggest you to at least implement the builtin correctly then.

This is indeed important and would allow you to test for the other targets.

Reply:
This is totally unnecessary work. We're trying to fix a problem in the parser and make the test pass.

Reply:
As we have a time constraint, please, write a comment there explaining the situation (i.e the builtin is not correctly implemented because we wanted the parser tests to work). I would accept your PR this way.

I have been thinking too if a `Instr::EVMUnimplemented` would make sense in this case, but I worry it would consume a little time to implement.

Reply:
I've given this some thought. In order to implement these yul builtins, we really need pointers in CFG. We could re-implement `Instr::Unreachable { ... }` as having a pointer and a length; I think that would make sense. However, for `Instr::Return { .. }` this is not so clear. What does mean if we do a yul return from a private function?

So I've gone down the route of `Instr::Unimplemented` for now. Maybe implementing an evm target will also help us understand better what to do with evm builtins on non-evm targets.